uTinyRipperCore/Parser/Classes/TrailRenderer/TrailRendererParamaters.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using uTinyRipper.AssetExporters;
+using uTinyRipper.Classes;
+using uTinyRipper.Classes.AnimationClips;
+using uTinyRipper.Classes.ParticleSystems;
+using uTinyRipper.SerializedFiles;
+using uTinyRipper.YAML;
+using Object = uTinyRipper.Classes.Object;
+
+namespace uTinyRipper.Parser.Classes.TrailRenderers
+{
+	public struct TrailRendererParameters : IAssetReadable, IYAMLExportable
+	{
+		public static int GetSerializedVersion(Version version)
+		{
+			if (version.IsGreaterEqual(2018, 3))
+			{
+				return 3;
+			}
+			if (version.IsGreaterEqual(5))
+			{
+				return 2;
+			}
+			return 1;
+		}
+		/// <summary>
+		/// 2019.2.0 and greater
+		/// </summary>
+		public static bool IsReadShadowBias(Version version)
+		{
+			return version.IsGreaterEqual(2018, 3);
+		}
+		public void Read(AssetReader reader)
+		{
+			WidthMultiplier = reader.ReadSingle();
+			WidthCurve.Read(reader);
+			ColorGradient.Read(reader);
+			NumCornerVertices = reader.ReadInt32();
+			NumCapVertices = reader.ReadInt32();
+			Alignment = (LineAlignment)reader.ReadInt32();
+			TextureMode = (LineTextureMode)reader.ReadInt32();
+			if (IsReadShadowBias(reader.Version))
+			{
+				ShadowBias = reader.ReadSingle();
+			}
+			GenerateLightingData = reader.ReadBoolean();
```

Comment:
This field was added in 2017.1.0b2

Reply:
Is the b in 2017.1.0b2 beta or base?

Reply:
beta. there is no such thing as base version. I didnt know that at the moment of creation this enum.base member for this enum is redunant